kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ResourceSelection.vue

```diff
@@ -490,9 +518,43 @@
           return total - numberOfQuestionsSelected;
         }
         return -1;
+      });
+
+      /** @returns {Boolean} Whether the given node should be displayed with a checkbox
+       *  @description Returns true for exercises or folders that have more than 0 unused questions
+       *  and adding it would give us fewer than maxSectionQuestionOptions questions
+       */
+      function nodeIsSelectableOrUnselectable(node) {
+        // For practice quizzes, we only want to allow selection of resources, not folders.
+        if (selectPracticeQuiz.value && node.kind === ContentNodeKinds.EXERCISE) {
+          return true;
+        }
+        if (selectPracticeQuiz.value) {
+          return false;
+        }
+        if (
+          contentPresentInWorkingResourcePool(node) ||
+          contentPartlyPresentInWorkingResourcePool(node)
+        ) {
+          // If a node has been selected or partly selected, always allow it to be deselected.
+          return true;
+        }
```

Comment:
This situation is weird:

- 12 questions "Number of questions"
- Select QA Channel > Exercises > CK12 Exercises (111 questions)
- Go back to QA Channel root, click the indeterminate checkbox on QA Channel
- Now I've been allowed to select 500+ questions, and I get an error and I have to click to deselect everything
- I shouldn't have been allowed to select that many questions with my current question count

I think maybe disabling the checkbox altogether might be better than this. 

Or maybe we could instead just deselect the relevant selection when there are too many questions to add -- but this might be confusing in its own right (ie, "why does it select things in some cases but deselect in others?" might not have an obvious answer.)

Reply:
Yes - I think I missed an edge case here and didn't ensure that checkboxes are always disabled for channels (even if we are displaying to show the descendant selector state).